adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m

```diff
@@ -0,0 +1,251 @@
+  if (error) {
+    _adLoadCompletionHandler(nil, error);
+  }
+}
+
+- (NSError *)renderAd {
+  VungleAdNetworkExtras *extras = (VungleAdNetworkExtras *)[_adConfiguration extras];
+  NSMutableDictionary *options = nil;
+  if (extras) {
+    options = [[NSMutableDictionary alloc] init];
+    if (extras.muteIsSet) {
+      GADMAdapterVungleMutableDictionarySetObjectForKey(options, VunglePlayAdOptionKeyStartMuted,
+                                                        @(extras.muted));
+    }
+    if (extras.userId) {
+      GADMAdapterVungleMutableDictionarySetObjectForKey(options, VunglePlayAdOptionKeyUser,
+                                                        extras.userId);
+    }
+    if (extras.flexViewAutoDismissSeconds) {
+      GADMAdapterVungleMutableDictionarySetObjectForKey(options,
+                                                        VunglePlayAdOptionKeyFlexViewAutoDismissSeconds,
+                                                        @(extras.flexViewAutoDismissSeconds));
+    }
+  }
+  NSError *bannerError = nil;
+  [VungleSDK.sharedSDK addAdViewToView:_bannerView
+                           withOptions:options
+                           placementID:self.desiredPlacement
+                              adMarkup:[self bidResponse]
+                                 error:&bannerError];
+  return bannerError;
+}
+
+- (void)cleanUp {
+  if (_didBannerFinishPresenting) {
+    return;
+  }
+  _didBannerFinishPresenting = YES;
+
+  [VungleSDK.sharedSDK finishDisplayingAd:self.desiredPlacement adMarkup:self.bidResponse];
+  [[GADMAdapterVungleBiddingRouter sharedInstance] removeDelegate:self];
+  _bannerView = nil;
+}
+
+#pragma mark - GADMAdapterVungleDelegate delegates
+
+- (NSString *)bidResponse {
+    return [_adConfiguration bidResponse];
+}
+
+- (GADAdSize)bannerAdSize {
+  return _bannerSize;
+}
+
+- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
+  if (!isSuccess) {
+    _adLoadCompletionHandler(nil, error);
+    return;
+  }
+  [self loadAd];
+}
+
+- (void)adAvailable {
+  if (_isAdLoaded) {
+    // Already invoked an ad load callback.
+    return;
+  }
+  _isAdLoaded = YES;
+  _bannerView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, _bannerSize.size.width, _bannerSize.size.height)];
+  self.bannerState = BannerRouterDelegateStateWillPlay;
+  NSError *error = [self renderAd];
+  if (error) {
+    _adLoadCompletionHandler(nil, error);
+    return;
+  }
+    
+  if (_adLoadCompletionHandler) {
+    _delegate = _adLoadCompletionHandler(self, nil);
+  }
+
+  if (!_delegate) {
+    [[GADMAdapterVungleBiddingRouter sharedInstance] removeDelegate:self];
+    return;
+  }
+}
+
+- (void)adNotAvailable:(nonnull NSError *)error {
+  if (_isAdLoaded) {
+    // Already invoked an ad load callback.
+    return;
+  }
+  _adLoadCompletionHandler(nil, error);
+}
+
+- (void)willShowAd {
+  self.bannerState = BannerRouterDelegateStatePlaying;
+}
+
+- (void)didViewAd {
+  // Do nothing.
```

Comment:
@stanley-vungle 
Shouldn't there be `  [_delegate reportImpression];` here?

Reply:
I realized `-reportImpression` isn't called in both banner and interstitial ads, is it intended?

Reply:
There is no intention to not include it here.

Reply:
Looking back, I believe that it wasn't included because the `GADMAdapterVungleDelegate` portion was imported from the waterfall version and that didn't have a impression call in that.